### PR TITLE
[app-ext] add Application Extension guide

### PR DESCRIPTION
**Pull request by stepanblyschak**
Signed-off-by: Stepan Blyschak <stepanb@nvidia.com>

### CODE REVIEW

File: doc/sonic-application-extension/sonic-application-extension-guide.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+The command line to execute tests is:
+
+```
+pytest-3 -v
+```
+
+### Developing a new SONiC Application Extension
+
+To develop a new SONiC Application Extension use the following example extension as a template:
+https://github.com/stepanblyschak/sonic-example-extension.
```
Comment (liat-grozovik) [inline doc/sonic-application-extension/sonic-application-extension-guide.md:106]: @stepanblyschak this reference to your private fork. not sure this is good approach. the example should be on public azure. should you consider to upstream it? need to talk to MSFT and decide on next step.

File: doc/sonic-application-extension/sonic-application-extension-guide.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+Prerequisites, build instructions and installation instructions are present in repository README.md file.
+
+### Adding 3rd party application to SONiC package database
+
+Modify files/build_templates/packages.json.j2 to include new package. Example:
+
+```json
+{
+    "cpu-report": {
```
Comment (liat-grozovik) [inline doc/sonic-application-extension/sonic-application-extension-guide.md:116]: @stepanblyschak  if you are referring to your example application i suggest you mention it

Comment (liat-grozovik) [reply doc/sonic-application-extension/sonic-application-extension-guide.md:116]: taking a look at https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/packages.json.j2, do you mean to add it at the end of the file?

Comment (liat-grozovik) [reply doc/sonic-application-extension/sonic-application-extension-guide.md:116]: is this done manually by the wjh extension for example?

File: doc/sonic-application-extension/sonic-application-extension-guide.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+```
+$(CPU_REPORT)_DEFAULT_FEATURE_STATE_ENABLED # "y" or "n" - whether feature is enabled by default at system start. Sets enabled in the FEATURE table. Disabled by default.
+$(CPU_REPORT)_DEFAULT_FEATURE_OWNER # "local" or "kube". Default is "local".
+```
+
+### Manifest Reference
+
+Label that manifest content should be written to:
+```
+com.azure.sonic.manifest
```
Comment (liat-grozovik) [inline doc/sonic-application-extension/sonic-application-extension-guide.md:148]: @stepanblyschak writing this is google leads to many options. which one are you referring to?

**Comment by liat-grozovik** [comment]
there are many references to your private forks or documentation.
this should be further fixed, right?

File: doc/sonic-application-extension/sonic-application-extension-guide.md (hunk 1 above)
Comment (stepanblyschak) [reply doc/sonic-application-extension/sonic-application-extension-guide.md:106]: Yes, correct. It is better to be on Azure. I will ask Volodymyr to add this topic.
I am not able to create repo's or PR to create a repo in Azure.

File: doc/sonic-application-extension/sonic-application-extension-guide.md (hunk 2 above)
Comment (stepanblyschak) [reply doc/sonic-application-extension/sonic-application-extension-guide.md:116]: > @stepanblyschak if you are referring to your example application i suggest you mention it

Added a reference to example application.



> taking a look at https://github.com/Azure/sonic-buildimage/blob/master/files/build_templates/packages.json.j2, do you mean to add it at the end of the file?

Technically, it is a JSON object (dictionary) - the order does not metter.



> is this done manually by the wjh extension for example?

We have to add wjh record to this file manually if we decide to go this way.

File: doc/sonic-application-extension/sonic-application-extension-guide.md (hunk 3 above)
Comment (stepanblyschak) [reply doc/sonic-application-extension/sonic-application-extension-guide.md:148]: This is the label name. It is in the format of reverse domain notation which is following https://docs.docker.com/config/labels-custom-metadata/#key-format-recommendations,

**Comment by stepanblyschak** [comment]
@liat-grozovik I fixed a link to HLD by making it relative link. This will point to Azure once it gets merged.

For the link to my private repository, we should ask MSFT to create one for example extension if they agree.

**Pull request closed without merging**